src/components/VmDetails/cards/SnapshotsCard/sagas.js

```diff
@@ -38,9 +55,11 @@ function* deleteVmSnapshot (action) {
 }
 
 function* restoreVmSnapshot (action) {
+  yield put(addSnapshotRestorePendingTask())
   yield startProgress({ vmId: action.payload.vmId, name: 'restoreSnapshot' })
   const result = yield callExternalAction('restoreSnapshot', Api.restoreSnapshot, action)
   yield stopProgress({ vmId: action.payload.vmId, name: 'restoreSnapshot', result })
+  yield put(removeSnapshotRestorePendingTask())
```

Comment:
Is restore sync so once the call returns it is done?  If not, the `delaySec` dance would need to be done here as well...